### PR TITLE
fix: enforce json schema for prompts and conversations

### PR DESCRIPTION
**Pull request by Maxim-Gadalov**
#213

### CODE REVIEW

File: src/main/java/com/epam/aidial/core/controller/ResourceController.java [hunk 1]
```diff
@@ -137,6 +140,14 @@ private Future<?> putResource(ResourceDescription descriptor) {
                     }
 
                     String body = bytes.toString(StandardCharsets.UTF_8);
+
+                    ResourceType resourceType = descriptor.getType();
+                    switch (resourceType) {
+                        case PROMPT -> ProxyUtil.convertToObject(body, Prompt.class);
+                        case CONVERSATION -> ProxyUtil.convertToObject(body, Conversation.class);
+                        default -> throw new IllegalArgumentException("Unsupported resource type " + resourceType);
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/controller/ResourceController.java:148]: It had better to throw HttpException
`throw new HttpException(HttpStatus.BAD_REQUEST, "Unsupported resource type " + resourceType);`

File: src/main/java/com/epam/aidial/core/controller/ResourceController.java [hunk 2]
```diff
@@ -149,6 +160,8 @@ private Future<?> putResource(ResourceDescription descriptor) {
                 .onFailure(error -> {
                     if (error instanceof HttpException exception) {
                         context.respond(exception.getStatus(), exception.getMessage());
+                    } else if (error instanceof IllegalArgumentException badRequest) {
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/controller/ResourceController.java:163]: See the previous comment

**Comment by astsiapanay** [comment]
May any fields be optional in the JSON schema? If so we shouldn't throw an exception and mark these fields optional.

**Comment by astsiapanay** [comment]
Should we ignore unknown fields?

File: src/main/java/com/epam/aidial/core/controller/ResourceController.java (hunk 2 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/controller/ResourceController.java:163]: see ProxyUtils.convertToObject

File: src/main/java/com/epam/aidial/core/controller/ResourceController.java (hunk 1 above)
Comment (Maxim-Gadalov) [reply src/main/java/com/epam/aidial/core/controller/ResourceController.java:148]: IllegalArgumentException will be caught and transformed to new HttpException(HttpStatus.BAD_REQUEST, "Unsupported resource type " + resourceType); later

**Review by astsiapanay — DISMISSED** [review]

**Comment by Maxim-Gadalov** [comment]
/deploy-review

>Environment URL: https://chat-ai-dial-core-pr-253.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Maxim-Gadalov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-253.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**